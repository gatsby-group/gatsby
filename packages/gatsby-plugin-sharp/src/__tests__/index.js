const plugin = require('../');

describe('gatsby-plugin-sharp', () => {
  it('contains a test', () => {

  });
});
