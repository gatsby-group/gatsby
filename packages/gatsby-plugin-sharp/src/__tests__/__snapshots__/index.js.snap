// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-sharp base64 converts image to base64 1`] = `
Object {
  "aspectRatio": 1,
  "height": 20,
  "originalName": "test.png",
  "src": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAABKklEQVQ4y8WUy07DMBBFrwMpr4aybjepUeqIFd/CAgmJsuIHgWXVL+Gx4C3xCdQJnpm0UDJNAQnVuouMM8dzY4+DcoA/C/8Fe9bv4ABMMpQOZc5yFPqfwEUQY499jHoY9/BsKQyTRTNMyzu8WAwTbBnI2DY438XbPr0qFsFCvlrkMTERYFgRL3HYUvh5OMdpQqkbBtPC9NDiINQPCZM6LGVvUzIpwNch4V5EGxHS/DeYtjfHZbcyXB8yGfaPimcafLEMHqmw2L5OsWkUUuY6Ee5V28WUP25TXjwPS3iWUFnfcFQPFna98rlmSGL4IOZucQvgT76Po53qeOTYTtp4qpFKe/qM29PhJsVVl3SXsluntLfS7rPvn10MP1Aau+lKBuA9I/kV/AyW6gNTIWTuA/r8lQAAAABJRU5ErkJggg==",
  "width": 20,
}
`;

exports[`gatsby-plugin-sharp duotone fixed 1`] = `
Object {
  "aspectRatio": 1,
  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAAH6ji2bAAAACXBIWXMAAAPoAAAD6AG1e1JrAAABk0lEQVQ4y7VUy07DMBDsf9IDhcauaVM4IajUIuALOFRI/R24wB8AgkpInJKI9ME5zDi24rx5HjZKvOP17Hg2neh8nNjo8CGESGL7sbpwMvrlaepriF4J3dQLMlsXvzbVFCKHLJ3J8+5PRxqlF0OzleFLkSG7Xrp4faiyRaI/Lgs13QhNVB5uCUyUTDyUloidvkjezvyMEB8B4uZkqAHCiR44bYtCcecSirD5XQDmIGzVjOrEiJo4fhvY2jWTG3C6g/TLmZ+8VwEJekWy20+loUSzA5kDp15BKJmXhmD6Iqcjtep5eSArk0ZQPPpqPNAaWiBpbKt0pLgEU2xuIOewTh7rmFWFRH9/M78uaG8yNuox4oYbbrQEhaI4HKEh7mvfS2OE98WR0r6P26zj2ofW5rWowlzY2WduAQxZ14odmNZuMWA0h6wo5pplDxj+MrgnqDM3W32E/QYNxaxN2f7ztNx6qWUCOC3H+F/Y35EU2XCw3QkGhBrHX/WXLbwG4wewoAwMst84t/0jH4ZG36DFLv9m7E/1HjvIa/canAAAAABJRU5ErkJggg==",
  "height": 100,
  "originalName": "test.png",
  "src": "/static/1234/2f4e8/test.png",
  "srcSet": "/static/1234/2f4e8/test.png 1x",
  "tracedSVG": undefined,
  "width": 100,
}
`;

exports[`gatsby-plugin-sharp duotone fluid 1`] = `
Object {
  "aspectRatio": 1,
  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAAH6ji2bAAAACXBIWXMAAAPoAAAD6AG1e1JrAAABk0lEQVQ4y7VUy07DMBDsf9IDhcauaVM4IajUIuALOFRI/R24wB8AgkpInJKI9ME5zDi24rx5HjZKvOP17Hg2neh8nNjo8CGESGL7sbpwMvrlaepriF4J3dQLMlsXvzbVFCKHLJ3J8+5PRxqlF0OzleFLkSG7Xrp4faiyRaI/Lgs13QhNVB5uCUyUTDyUloidvkjezvyMEB8B4uZkqAHCiR44bYtCcecSirD5XQDmIGzVjOrEiJo4fhvY2jWTG3C6g/TLmZ+8VwEJekWy20+loUSzA5kDp15BKJmXhmD6Iqcjtep5eSArk0ZQPPpqPNAaWiBpbKt0pLgEU2xuIOewTh7rmFWFRH9/M78uaG8yNuox4oYbbrQEhaI4HKEh7mvfS2OE98WR0r6P26zj2ofW5rWowlzY2WduAQxZ14odmNZuMWA0h6wo5pplDxj+MrgnqDM3W32E/QYNxaxN2f7ztNx6qWUCOC3H+F/Y35EU2XCw3QkGhBrHX/WXLbwG4wewoAwMst84t/0jH4ZG36DFLv9m7E/1HjvIa/canAAAAABJRU5ErkJggg==",
  "density": 72,
  "originalImg": "/static/1234/2f4e8/test.png",
  "originalName": "test.png",
  "presentationHeight": 100,
  "presentationWidth": 100,
  "sizes": "(max-width: 100px) 100vw, 100px",
  "src": "/static/1234/2f4e8/test.png",
  "srcSet": "/static/1234/e3e32/test.png 25w,
/static/1234/281bc/test.png 50w,
/static/1234/2f4e8/test.png 100w",
  "srcSetType": "image/png",
  "tracedSVG": undefined,
}
`;

exports[`gatsby-plugin-sharp fixed correctly infers the width when only the height is given 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "height": 10,
                "pathPrefix": "",
                "toFormat": "png",
              },
              "outputPath": "fb009/test.png",
            },
            Object {
              "args": Object {
                "height": 15,
                "pathPrefix": "",
                "toFormat": "png",
              },
              "outputPath": "0314c/test.png",
            },
            Object {
              "args": Object {
                "height": 20,
                "pathPrefix": "",
                "toFormat": "png",
              },
              "outputPath": "24504/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fixed does not warn when the requested width is equal to the image width 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 1,
              },
              "outputPath": "22212/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 2,
              },
              "outputPath": "1fbec/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 2,
              },
              "outputPath": "1fbec/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/test.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fixed should give the same result with createJob as with createJobV2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 1,
              },
              "outputPath": "22212/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 2,
              },
              "outputPath": "1fbec/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 2,
              },
              "outputPath": "1fbec/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/test.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fixed warns when the requested width is greater than the image width 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 100,
              },
              "outputPath": "62ab9/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/test.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid accepts srcSet breakpoints 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 50,
              },
              "outputPath": "2f62a/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 70,
              },
              "outputPath": "543b9/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 150,
              },
              "outputPath": "55f83/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 250,
              },
              "outputPath": "e633e/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 281,
              },
              "outputPath": "5bca6/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid adds pathPrefix if defined 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "/blog",
                "toFormat": "png",
                "width": 100,
              },
              "outputPath": "62ab9/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/test.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid calculate height based on width when maxWidth & maxHeight are present 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "height": 5,
                "pathPrefix": "",
                "toFormat": "png",
                "width": 5,
              },
              "outputPath": "6931e/test.png",
            },
            Object {
              "args": Object {
                "height": 10,
                "pathPrefix": "",
                "toFormat": "png",
                "width": 10,
              },
              "outputPath": "dba9b/test.png",
            },
            Object {
              "args": Object {
                "height": 20,
                "pathPrefix": "",
                "toFormat": "png",
                "width": 20,
              },
              "outputPath": "27e70/test.png",
            },
            Object {
              "args": Object {
                "height": 30,
                "pathPrefix": "",
                "toFormat": "png",
                "width": 30,
              },
              "outputPath": "1fff3/test.png",
            },
            Object {
              "args": Object {
                "height": 40,
                "pathPrefix": "",
                "toFormat": "png",
                "width": 40,
              },
              "outputPath": "3d276/test.png",
            },
            Object {
              "args": Object {
                "height": 281,
                "pathPrefix": "",
                "toFormat": "png",
                "width": 281,
              },
              "outputPath": "79217/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid does not change the arguments object it is given 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 100,
              },
              "outputPath": "62ab9/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/test.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid ensure maxWidth is in srcSet breakpoints 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 50,
              },
              "outputPath": "2f62a/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 70,
              },
              "outputPath": "543b9/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 150,
              },
              "outputPath": "55f83/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 200,
              },
              "outputPath": "5da2d/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 281,
              },
              "outputPath": "5bca6/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid includes responsive image properties, e.g. sizes, srcset, etc. 1`] = `
Object {
  "aspectRatio": 1,
  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAABKklEQVQ4y8WUy07DMBBFrwMpr4aybjepUeqIFd/CAgmJsuIHgWXVL+Gx4C3xCdQJnpm0UDJNAQnVuouMM8dzY4+DcoA/C/8Fe9bv4ABMMpQOZc5yFPqfwEUQY499jHoY9/BsKQyTRTNMyzu8WAwTbBnI2DY438XbPr0qFsFCvlrkMTERYFgRL3HYUvh5OMdpQqkbBtPC9NDiINQPCZM6LGVvUzIpwNch4V5EGxHS/DeYtjfHZbcyXB8yGfaPimcafLEMHqmw2L5OsWkUUuY6Ee5V28WUP25TXjwPS3iWUFnfcFQPFna98rlmSGL4IOZucQvgT76Po53qeOTYTtp4qpFKe/qM29PhJsVVl3SXsluntLfS7rPvn10MP1Aau+lKBuA9I/kV/AyW6gNTIWTuA/r8lQAAAABJRU5ErkJggg==",
  "density": 72,
  "originalImg": "/static/1234/62ab9/test.png",
  "originalName": "test.png",
  "presentationHeight": 100,
  "presentationWidth": 100,
  "sizes": "(max-width: 100px) 100vw, 100px",
  "src": "/static/1234/62ab9/test.png",
  "srcSet": "/static/1234/62ab9/test.png 100w",
  "srcSetType": "image/png",
  "tracedSVG": undefined,
}
`;

exports[`gatsby-plugin-sharp fluid includes responsive image properties, e.g. sizes, srcset, etc. with the createJob api 1`] = `
Object {
  "aspectRatio": 1,
  "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAAACXBIWXMAAAsSAAALEgHS3X78AAABKklEQVQ4y8WUy07DMBBFrwMpr4aybjepUeqIFd/CAgmJsuIHgWXVL+Gx4C3xCdQJnpm0UDJNAQnVuouMM8dzY4+DcoA/C/8Fe9bv4ABMMpQOZc5yFPqfwEUQY499jHoY9/BsKQyTRTNMyzu8WAwTbBnI2DY438XbPr0qFsFCvlrkMTERYFgRL3HYUvh5OMdpQqkbBtPC9NDiINQPCZM6LGVvUzIpwNch4V5EGxHS/DeYtjfHZbcyXB8yGfaPimcafLEMHqmw2L5OsWkUUuY6Ee5V28WUP25TXjwPS3iWUFnfcFQPFna98rlmSGL4IOZucQvgT76Po53qeOTYTtp4qpFKe/qM29PhJsVVl3SXsluntLfS7rPvn10MP1Aau+lKBuA9I/kV/AyW6gNTIWTuA/r8lQAAAABJRU5ErkJggg==",
  "density": 72,
  "originalImg": "/static/1234/62ab9/test.png",
  "originalName": "test.png",
  "presentationHeight": 100,
  "presentationWidth": 100,
  "sizes": "(max-width: 100px) 100vw, 100px",
  "src": "/static/1234/62ab9/test.png",
  "srcSet": "/static/1234/62ab9/test.png 100w",
  "srcSetType": "image/png",
  "tracedSVG": undefined,
}
`;

exports[`gatsby-plugin-sharp fluid infers the maxWidth if only maxHeight is given 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "height": 5,
                "pathPrefix": "",
                "toFormat": "png",
                "width": undefined,
              },
              "outputPath": "c2bf1/test.png",
            },
            Object {
              "args": Object {
                "height": 10,
                "pathPrefix": "",
                "toFormat": "png",
                "width": undefined,
              },
              "outputPath": "b9e31/test.png",
            },
            Object {
              "args": Object {
                "height": 20,
                "pathPrefix": "",
                "toFormat": "png",
                "width": undefined,
              },
              "outputPath": "f589b/test.png",
            },
            Object {
              "args": Object {
                "height": 30,
                "pathPrefix": "",
                "toFormat": "png",
                "width": undefined,
              },
              "outputPath": "e2b80/test.png",
            },
            Object {
              "args": Object {
                "height": 40,
                "pathPrefix": "",
                "toFormat": "png",
                "width": undefined,
              },
              "outputPath": "5760a/test.png",
            },
            Object {
              "args": Object {
                "height": 136,
                "pathPrefix": "",
                "toFormat": "png",
                "width": undefined,
              },
              "outputPath": "31b03/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid keeps original file name 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 100,
              },
              "outputPath": "62ab9/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/test.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid prevents duplicate breakpoints 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 50,
              },
              "outputPath": "2f62a/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 100,
              },
              "outputPath": "62ab9/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 250,
              },
              "outputPath": "e633e/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 281,
              },
              "outputPath": "5bca6/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid reject any breakpoints larger than the original width 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 50,
              },
              "outputPath": "2f62a/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 70,
              },
              "outputPath": "543b9/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 150,
              },
              "outputPath": "55f83/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 250,
              },
              "outputPath": "e633e/test.png",
            },
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 281,
              },
              "outputPath": "5bca6/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid should give the same result with createJob as with createJobV2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "pathPrefix": "",
                "toFormat": "png",
                "width": 100,
              },
              "outputPath": "62ab9/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/test.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp fluid should throw if maxWidth is less than 1 1`] = `[MockFunction]`;

exports[`gatsby-plugin-sharp fluid should throw on srcSet breakpoints less than 1 1`] = `[MockFunction]`;

exports[`gatsby-plugin-sharp image quirks handles padding bytes correctly 1`] = `
Object {
  "hUnits": "px",
  "height": 1000,
  "mime": "image/jpeg",
  "type": "jpg",
  "wUnits": "px",
  "width": 746,
}
`;

exports[`gatsby-plugin-sharp queueImageResizing with createJob file name works with spaces & special characters createJob 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "toFormat": "png",
                "width": 3,
              },
              "outputPath": "2cee9/spaces and '\\"@#$%^&,.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp queueImageResizing with createJob should round height when auto-calculated createJob 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "toFormat": "png",
                "width": 3,
              },
              "outputPath": "2cee9/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
      Object {},
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp queueImageResizing with createJobV2 file name works with spaces & special characters createJobV2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "toFormat": "png",
                "width": 3,
              },
              "outputPath": "2cee9/spaces and '\\"@#$%^&,.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp queueImageResizing with createJobV2 should round height when auto-calculated createJobV2 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "args": Object {
          "operations": Array [
            Object {
              "args": Object {
                "toFormat": "png",
                "width": 3,
              },
              "outputPath": "2cee9/test.png",
            },
          ],
          "pluginOptions": Object {
            "defaultQuality": 50,
            "forceBase64Format": false,
            "lazyImageGeneration": true,
            "stripMetadata": true,
            "useMozJpeg": false,
          },
        },
        "inputPaths": Array [
          "<PROJECT_ROOT>/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png",
        ],
        "name": "IMAGE_PROCESSING",
        "outputDir": "<PROJECT_ROOT>/public/static/1234",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`gatsby-plugin-sharp stats determines if the image is transparent, based on the presence and use of alpha channel 1`] = `
Object {
  "isTransparent": false,
}
`;

exports[`gatsby-plugin-sharp stats determines if the image is transparent, based on the presence and use of alpha channel 2`] = `
Object {
  "isTransparent": true,
}
`;

exports[`gatsby-plugin-sharp tracedSVG runs on demand 1`] = `
Object {
  "aspectRatio": 1,
  "base64": undefined,
  "height": 100,
  "originalName": "test.png",
  "src": "/static/1234/62ab9/test.png",
  "srcSet": "/static/1234/62ab9/test.png 1x",
  "tracedSVG": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='100'%20height='100'%3e%3cpath%20d='M41%2024c-18%207-24%2029-11%2043%2015%2017%2044%208%2046-15%201-19-17-34-35-28'%20fill='red'%20fill-rule='evenodd'/%3e%3c/svg%3e",
  "width": 100,
}
`;

exports[`gatsby-plugin-sharp tracedSVG runs on demand 2`] = `
Object {
  "aspectRatio": 1,
  "base64": undefined,
  "density": 72,
  "originalImg": "/static/1234/62ab9/test.png",
  "originalName": "test.png",
  "presentationHeight": 100,
  "presentationWidth": 100,
  "sizes": "(max-width: 100px) 100vw, 100px",
  "src": "/static/1234/62ab9/test.png",
  "srcSet": "/static/1234/bb8d1/test.png 25w,
/static/1234/2f62a/test.png 50w,
/static/1234/62ab9/test.png 100w",
  "srcSetType": "image/png",
  "tracedSVG": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='100'%20height='100'%3e%3cpath%20d='M41%2024c-18%207-24%2029-11%2043%2015%2017%2044%208%2046-15%201-19-17-34-35-28'%20fill='red'%20fill-rule='evenodd'/%3e%3c/svg%3e",
}
`;
